.github/workflows/rust.yml

```diff
@@ -158,6 +158,27 @@ jobs:
       run: cargo check-all-features --workspace --all-targets
 
 
+  # NOTE: the term "smoke test" comes from electronics design: the minimal
+  # expectations anyone has in their device is to not catch fire on boot.
+  smoke-no-std:
+    name: Make sure no-std works
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@1.69.0
+      with:
+        targets: wasm32-unknown-unknown
+    - name: Set up cargo cache
+      uses: Swatinem/rust-cache@v2
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Check
+      run: |
+        cd ensure_no_std
+        cargo build --release
+        cargo build --release -F alloc
+
```

Comment:
I think this should be covered by the `build-programs` job.

Reply:
I don't understand what you mean. The `build-programs` job compiles Cairo programs. Did you perhaps mean `build-release`? I left this one in another job mainly because of the `wasm32-unknown-unknown` target. Should we merge them?

Reply:
Done in a732598